nixos/modules/services/web-apps/keycloak.nix

```diff
@@ -752,41 +598,37 @@ in
                 "mysql.service"
               ]
               else [ ];
+            secretPaths = catAttrs "_secret" (collect isSecret cfg.settings);
+            mkSecretReplacement = file: ''
+              replace-secret ${hashString "sha256" file} $CREDENTIALS_DIRECTORY/${baseNameOf file} /run/keycloak/conf/keycloak.conf
+            '';
+            secretReplacements = lib.concatMapStrings mkSecretReplacement secretPaths;
```

Comment:
Would it be easier to use the `vault-dir` option, rather than providing linking to secrets here and there?

Reply:
I see the value- I'm a bit leery of introducing a particular pattern for a few modules as one of many, but we need something, and shouldn't block changes until there's strong consensus..

Reply:
Yeah, it's already a pattern and has been for quite some time, but it would be nice to standardize it, turn it into a type instead of a pattern, and make it super easy to implement in modules.